### DISCUSSION TITLE
Why let is used instead of const?

### DISCUSSION THREAD

**Discussion by pvorona** [adobe/react-spectrum#6414] [General]
Let is used across the source even though variables don't require reassignment. What's the reason of using let instead of const?

https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/ColorField.tsx
https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Button.tsx

**Comment by snowystinger** [comment]
We find that it provides a false sense of security and causes more churn in PRs.
Specifically that const !== immutable
```
const a = {b: 12};
function doNothingBad(p) {
  p.b = NaN;
}
doNothingBad(a);
console.log(p);
```
You can see https://www.reddit.com/r/javascript/comments/cfaayg/askjs_lets_debate_about_this_article_const_vs_let/ for more information about this whole debate.

We realize this is an opinion, and not everyone will agree with what we've decided to do in our codebase. Indeed, contributors frequently add `const`, and sometimes they get into the codebase. We'll change it if we ever need to.
Our team decided this was the best decision for our work.
